agent/stats/engine.go

```diff
@@ -296,13 +318,109 @@ func (engine *DockerStatsEngine) GetInstanceMetrics() (*ecstcs.MetricsMetadata,
 	return metricsMetadata, taskMetrics, nil
 }
 
+// GetTaskHealthMetrics returns the container health metrics
+func (engine *DockerStatsEngine) GetTaskHealthMetrics() (*ecstcs.HealthMetadata, []*ecstcs.TaskHealth, error) {
+	var taskHealths []*ecstcs.TaskHealth
+	idle := engine.isHealthContainerIdle()
+	metadata := &ecstcs.HealthMetadata{
+		Cluster:           aws.String(engine.cluster),
+		ContainerInstance: aws.String(engine.containerInstanceArn),
+		Fin:               aws.Bool(idle),
+		MessageId:         aws.String(uuid.NewRandom().String()),
+	}
+
+	if idle {
+		seelog.Debug("Instance is idle. No task health metrics to report")
+		return metadata, taskHealths, nil
+	}
+
+	engine.lock.RLock()
+	defer engine.lock.RUnlock()
+
+	for taskARN := range engine.tasksToHealthCheckContainers {
+		taskHealth := engine.getTaskHealthUnsafe(taskARN)
+		if taskHealth == nil {
+			continue
+		}
+		taskHealths = append(taskHealths, taskHealth)
+	}
+
+	if len(taskHealths) == 0 {
+		return nil, nil, EmptyMetricsError
```

Comment:
please return a different error here. `EmptyMetricsError` is not the same as `EmptyHealthStatusError` for example